bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -405,7 +405,27 @@ public boolean resumeSuspendedClients(final int numberOfInMemBlockPermits)
 
   public boolean isMemoryBlockAvailable()
   {
-    return (storage == null) || (numberOfInMemBlockPermits.get() > 0);
+    return (numberOfInMemBlockPermits.get() > 0) || ((storage == null) && !areSubscribersBehindByMax());
+  }
+
+  private boolean areSubscribersBehindByMax()
+  {
+    boolean behind = false;
+    if (backPressureEnabled) {
+      int count = 0;
+      for (Block curr = last; curr != null; curr = curr.prev, count++) {
```

Comment:
Are you thinking about scenarios when list is being structurally modified by purge, for example. Even in those cases it should not result in wrong results (as there will be no subscribers in those blocks) on NPEs?